src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -394,6 +400,62 @@ yield return
                 }
             }
 
+            private StatementSyntax RemoveOutVariablesFromDeclarationExpressions(StatementSyntax statement,
+                HashSet<string> variablesToRemove)
+            {
+                var replacements = PooledDictionary<SyntaxNode, SyntaxNode>.GetInstance();
+
+                var declarations = statement.DescendantNodes()
+                    .Where(n => n.IsKind(SyntaxKind.DeclarationExpression, SyntaxKind.DeclarationPattern));
+
+                foreach (var node in declarations)
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.DeclarationExpression:
+                            var declaration = (DeclarationExpressionSyntax)node;
+                            if (declaration.VariableComponent.Kind() != SyntaxKind.TypedVariableComponent)
+                            {
+                                break;
+                            }
+
+                            var variableComponent = (TypedVariableComponentSyntax)declaration.VariableComponent;
+                            if (variableComponent.Designation.Kind() != SyntaxKind.SingleVariableDesignation)
+                            {
+                                break;
+                            }
+
+                            var designation = (SingleVariableDesignationSyntax)variableComponent.Designation;
+                            var name = designation.Identifier.ValueText;
+                            if (variablesToRemove.Contains(name))
+                            {
+                                replacements.Add(declaration, SyntaxFactory.IdentifierName(name));
+                            }
+
+                            break;
+
+                        case SyntaxKind.DeclarationPattern:
+                            var pattern = (DeclarationPatternSyntax)node;
+                            if (!variablesToRemove.Contains(pattern.Identifier.ValueText))
+                            {
+                                break;
+                            }
+
+                            // We don't have a good refactoring for this, so we just annotate the conflict
+                            var identifier = pattern.Identifier;
+                            var annotation = ConflictAnnotation.Create(CSharpFeaturesResources.Conflict_s_detected);
+                            var newIdentifier = identifier.WithAdditionalAnnotations(annotation);
+                            replacements.Add(pattern, pattern.WithIdentifier(newIdentifier));
+
+                            break;
+                    }
+                }
+
+                var result = statement.ReplaceNodes(replacements.Keys, (orig, partiallyReplaced) => replacements[orig]);
```

Comment:
It seems `SyntaxEditor` needs a workspace, but as far as I can tell, code generators don't have one.  Rather than trying to thread one through, I think the Dictionary approach is reasonably simple.